packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/ArbTokenBridge.sol

```diff
@@ -86,35 +87,94 @@ contract ArbTokenBridge is CloneFactory {
         l1Pair = _l1Pair;
     }
 
+    function executePostMintCall(
+        bytes memory postMintCall
+    ) internal returns (bool, address) {
+        // TODO: should first bit define if handleCallFail is called?
+        if(postMintCall.length < 40) return (false, address(0));
+
+        address backupAddr = BytesLib.toAddress(postMintCall, 0);
+        address destAddr = BytesLib.toAddress(postMintCall, 20);
+
+        bool success;
+        assembly {
+            success := call(
+                gas(),
+                destAddr,
+                callvalue(),
+                add(postMintCall, 72), // 32 bytes rlp encoded bytes param + 40 bytes addresses
+                sub(postMintCall, 40),
+                0,
+                0
+            )
+        }
+
+        // TODO: what if user tries calling bridge mint from postMintCall?
+        // Set queryable variable and check if call originates from mint.
+        // TODO: Add this check to the token
+        return (success, backupAddr);
+    }
+
+    function handleCallFail(
+        address token,
+        address account,
+        address backupAddr,
+        uint256 amount
+    ) internal {
+        // if user sent to bridge, then proxied a call that failed, send back to user
+        if(
+            account == address(this) &&
+            backupAddr != address(0) &&
+            backupAddr != account // I don't think this check is needed
+        ) {
+            // TODO: if this fails?
+            bool success = IERC20(token).transfer(backupAddr, amount);
+        } else {
+            // damn
+        }
+    }
+
     function mintERC777FromL1(
         address l1ERC20,
         address account,
         uint256 amount,
-        uint8 decimals
+        uint8 decimals,
+        bytes calldata postMintCall
     ) external onlyEthPair {
         IArbToken token = ensureERC777TokenExists(l1ERC20, decimals);
         token.bridgeMint(account, amount);
+
+        (bool success, address backupAddr) = executePostMintCall(postMintCall);
+        if(!success) handleCallFail(address(token), account, backupAddr, amount);
     }
 
     function mintERC20FromL1(
         address l1ERC20,
         address account,
         uint256 amount,
-        uint8 decimals
+        uint8 decimals,
+        bytes calldata postMintCall
     ) external onlyEthPair {
         IArbToken token = ensureERC20TokenExists(l1ERC20, decimals);
         token.bridgeMint(account, amount);
+
+        (bool success, address backupAddr) = executePostMintCall(postMintCall);
```

Comment:
This opens up a potential attack.

Imagine if the user deposits a token, sets the `account` as the address of another bridged token contract, and sets the `postMintCall` to the data to mint infinite tokens to the attacker.